### ISSUE TITLE
Indent rule is not working correctly.

### ISSUE THREAD

**Issue by omeryagmurlu** [eslint/eslint#6927]
![problem](https://cloud.githubusercontent.com/assets/8201544/17752027/a584f3d6-64d2-11e6-8400-b0501cc634f7.png)

- On eslint v3.3.0.
- Using parser `babel-eslint`.

- Configuration: 
```js
module.exports = {
	rules: {
		'quotes': ['error', 'single', {
				avoidEscape: true
		}],
		'strict': 'warn',
		'max-len': ['warn', {
			code: 160,
			comments: 80
		}],
		'no-return-assign': ['error', 'except-parens'],
		'space-before-function-paren': ['error', 'never']
	},
	extends: 'xo/esnext',
	root: true
};
```
Related part of `xo` config:
```js
...
'indent': ['error', 'tab', {
	SwitchCase: 1
}],
...
```
- Code:
```js
export const pretest = () => gulp.src('lib/**/*.js')
    .pipe(_excludeGitignore())
    .pipe(_istanbul({
		includeUntested: true,
		instrumenter: _isparta.Instrumenter
	}))
    .pipe(_istanbul.hookRequire());
```

- Expected errors not to exist.

- Actual error message:
```sh
$ node_modules/eslint/bin/eslint.js gulpfile.babel.js

/home/../gulpfile.babel.js
  37:3  error  Expected indentation of 1 tab character but found 2   indent
  38:3  error  Expected indentation of 1 tab character but found 2   indent
  39:2  error  Expected indentation of 0 tab characters but found 1  indent

✖ 3 problems (3 errors, 0 warnings)
```

**Comment by gyandeeps** [comment]
Can you try your code without using `babel-eslint`? (with default parser).
And you can try the code which is in question here.

**Comment by omeryagmurlu** [comment]
Eslint gave me same errors using the default parser, no change.

And I didn't get what you meant by "the code which is in questions here" :smile:

**Comment by gyandeeps** [comment]
I can reproduce this issue with `default` parser. Are you sure you have tabs?

Test file:

```js
/* eslint indent: [2, "tab", {SwitchCase: 1}] */
export const pretest = () => gulp.src('lib/**/*.js')
--->.pipe(_excludeGitignore())
--->.pipe(_istanbul({
--->--->includeUntested: true,
--->--->instrumenter: _isparta.Instrumenter
--->}))
--->.pipe(_istanbul.hookRequire());
```


`--->` means 1 tab.

Run:

```sh
$ ./node_modules/.bin/eslint test.js --no-eslintrc --parser-options '{ecmaVersion: 6, sourceType: "module"}'

```

**Comment by vitorbal** [comment]
I also tried to reproduce this and could not. Here's my test file and config file:

```javascript
// All spaces are tabs
export const pretest = () => gulp.src('lib/**/*.js')
	.pipe(_excludeGitignore())
	.pipe(_istanbul({
		includeUntested: true,
		instrumenter: _isparta.Instrumenter
	}))
	.pipe(_istanbul.hookRequire());

```

```json
{
    "root": true,
    "parserOptions": {
        "sourceType": "module",
        "ecmaVersion": 6
    },
    "rules": {
        "indent": ["error", "tab", {
            "SwitchCase": 1
        }]
    }
}
```

```shell
$ node_modules/.bin/eslint --version
v3.3.1
$ node_modules/.bin/eslint test.js
# no errors
```

**Comment by omeryagmurlu** [comment]
Apparently line 36 was 4 spaces. Fixing it solved the problem.
